File: javascript-solutions/count-binary-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var countBinarySubstrings = function (s) {
	let prev = 0,
		ans = 0,
		curr = 1;

	for (let i = 1; i < s.length; i++) {
		if (s[i - 1] !== s[i]) {
			ans += Math.min(prev, curr);
			prev = curr;
			curr = 1;
		} else {
			curr += 1;
		}
	}
	return ans + Math.min(prev, curr);
};

File: python-solutions/count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        prevGroupCount = 0
        currGroupCount = 1
        ans = 0

        for i in range(1, len(s)):
            if s[i-1] != s[i]:
                ans += min(prevGroupCount, currGroupCount)
                prevGroupCount, currGroupCount = currGroupCount, 1
            else:
                currGroupCount += 1

        return ans + min(prevGroupCount, currGroupCount)
